### DISCUSSION TITLE
Is it possible to modify a distributed table engine of an existed clickhouse table?

### DISCUSSION THREAD

**Discussion by athree** [ClickHouse/ClickHouse#38253] [Q&A] [answered]
is it possible to modify a distributed table enable of an existed clickhouse table, while just modify cluster, for example:

update from :
CREATE TABLE xxx.yyy_distributed
(
    `key` String,
    `timestamp` DateTime
)
ENGINE = Distributed('clusterAAAA', 'xxx', 'yyy', rand()) 

to :
CREATE TABLE xxx.yyy_distributed
(
    `key` String,
    `timestamp` DateTime
)
ENGINE = Distributed('clusterBBB', 'xxx', 'yyy', rand()) 

if possible, where should i sync modify except config-file?

thx!!!

**Comment by UnamedRus** [comment, marked as answer]
You can just recreate Distributed table with new cluster definition or rename / exchange if you insert data in them.

1. RENAME TABLE yyy_distributed TO yyy_distributed_old, yyy_distributed_new TO yyy_distributed; OR (EXCHANGE TABLES yyy_distributed AND yyy_distributed_new)
2. Wait until all files from distributed table will be send to shards.
3. Drop old distributed table

**Reply by athree** [reply]
ths, i will try it!
